ValvePak/ValvePak/Package.Save.cs

```diff
@@ -1,13 +1,25 @@
+using Microsoft.VisualBasic.FileIO;
 using System;
 using System.Buffers;
+using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.IO.Hashing;
+using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
+using System.Text.RegularExpressions;
+using static System.Runtime.InteropServices.JavaScript.JSType;
 
 namespace SteamDatabase.ValvePak
 {
+	internal sealed class WriteEntry(ushort archiveIndex, uint fileOffset, PackageEntry entry)
```

Comment:
I don't think this is needed. You can calculate the ArchiveIndex directly in `AddFile`.

You can look at Valve's packedstore.cpp to see how they handle adding files:
- CPackedStore::AddFile has a bMultiChunk bool.
- They keep track of m_nHighestChunkFileIndex and then increase it if the file offset is higher than m_nWriteChunkSize which defaults to 200 * 1024 * 1024 bytes.

Reply:
That sounds good,I should go take a look at packdstore.cpp,Can you tell me where it is?

Reply:
Search for cstrike15_src

Reply:
I found it, thank you